bluetooth/src/iosMain/kotlin/scanner/Scanner.kt

```diff
@@ -0,0 +1,220 @@
+            info(TAG, "Did Discover Peripheral ${didDiscoverPeripheral.identifier.UUIDString}")
+            scanner.discoverPeripheral(central, didDiscoverPeripheral, advertisementData.typedMap(), RSSI.intValue)
+        }
+
+        fun handlePeripheral(didConnectPeripheral: CBPeripheral, block:suspend BaseDeviceConnectionManager.() -> Unit) = mainContinuation {
+            scanner.stateRepo.launchUseState { scannerState ->
+                if (scannerState is ScanningState.Initialized.Enabled)
+                    scannerState.discovered.devices.find { it.identifier == didConnectPeripheral.identifier }
+                        ?.let { device ->
+                            device.useState { deviceState ->
+                                block(deviceState.connectionManager)
+                            }
+                        }
+
+            }
+        }.invoke()
+
+        override fun centralManager(central: CBCentralManager, didConnectPeripheral: CBPeripheral) =
+            handlePeripheral(didConnectPeripheral) { handleConnect() }
+
+        override fun centralManager(central: CBCentralManager, didDisconnectPeripheral: CBPeripheral, error: NSError?) =
+            handlePeripheral(didDisconnectPeripheral) { handleDisconnect() }
+
+        override fun centralManager(central: CBCentralManager, didFailToConnectPeripheral: CBPeripheral, error: NSError?) =
+            handlePeripheral(didFailToConnectPeripheral) { handleDisconnect() }
+    }
+
+
+    private lateinit var mainCentralManager: CBCentralManager
+    private lateinit var checkEnabledCentralManager: CBCentralManager
+    private val centralManagers = emptyList<CBCentralManager>().toMutableList()
+//    private var connectionManagerMap = emptyMap<Identifier, BaseDeviceConnectionManager>().toMutableMap()
+    private val discoveringDelegates = mutableListOf<CBCentralManagerDelegateProtocol>()
+    private val activeDelegates = mutableSetOf<CBCentralManagerDelegateProtocol>()
+
+    private fun initMainManagers() {
+        if (!::mainCentralManager.isInitialized) {
+            mainCentralManager = CBCentralManager(null, dispatch_get_main_queue(), emptyMap<Any?, Any>())
+        }
+        if (!::checkEnabledCentralManager.isInitialized) {
+            checkEnabledCentralManager = CBCentralManager(null, dispatch_get_main_queue(), emptyMap<Any?, Any>())
+        }
+    }
+
+    override suspend fun scanForDevices(filter: Set<UUID>) {
+        if (filter.isEmpty()) {
+            val centralManager = CBCentralManager(null, dispatch_get_main_queue())
+            centralManagers.add(centralManager)
+            val awaitPoweredOn = EmptyCompletableDeferred()
+            val delegate = PoweredOnCBCentralManagerDelegate(this, awaitPoweredOn)
+            discoveringDelegates.add(delegate)
+            centralManager.delegate = delegate
+            awaitPoweredOn.await()
+            centralManager.scanForPeripheralsWithServices(null, null)
+        }
+
+        filter.forEach {
```

Comment:
This `forEach` and `if` before can be combine to avoid many duplicate lines

Reply:
I kept the logic the same (one central manager for each filter), but reduced the code. Might be worth it to see if all these central managers are needed though so I'll keep this open.